vue_firebase_option/src/views/Dashboard.vue

```diff
@@ -24,33 +30,40 @@ export default {
         uid: '',
       },
       balance: 0,
+      alertMessage: '',
+      isError: false,
     }
   },
   created() {
 
-    firebase.auth().onAuthStateChanged((currentUser) => {
-      if (currentUser) {
-          const user = this.$store.getters['user/user']
-          if (!user) {
-            this.authInfo.user = currentUser.displayName
-          } else {
-            this.authInfo.user = user
-          }
-
-          this.authInfo.uid = currentUser.uid
-          this.authInfo.email = currentUser.email
-          this.$store.dispatch('user/sign', this.authInfo)
+    const currentUser = firebase.auth().currentUser
+    const user = this.$store.getters['user/user']
+    if (!user) {
+      this.authInfo.user = currentUser.displayName
+    } else {
+      this.authInfo.user = user
+    }
 
-          const docRef = db.collection('users').doc(this.authInfo.uid)
-          docRef.get().then(doc => {
-            this.balance = doc.data().balance
-          })
+    this.authInfo.uid = currentUser.uid
+    this.authInfo.email = currentUser.email
+    this.$store.dispatch('user/sign', this.authInfo)
 
-      } else {
-        this.$router.push('/signin')
-      }
+    const docRef = db.collection('users').doc(this.authInfo.uid)
+    docRef.get().then(doc => {
+      this.balance = doc.data().balance
     })
+    .catch(() => {
+      this.alertMessage = 'Cannot get balance. Please reload!'
+      this.isError = true
+    });
   },
+  methods: {
+    signOut() {
+      firebase.auth().signOut().then(() => {
+        this.$router.push('/signin')
+      })
```

Comment:
この処理であれば、アロー関数を使う必要はないですね。
```suggestion
      firebase.auth().signOut().then(
        this.$router.push('/signin')
      )
```